src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedValuesFromArray = '';
+
+    for (let index = 0; index < this.length; index++) {
+      if (this[index] !== null && this[index] !== undefined) {
+        joinedValuesFromArray += this[index];
+      }
+
+      if (index < this.length - 1) {
```

Comment:
I suggest putting this.length in a variable, since we already use it twice, but I guess it's optional